core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeExternalSorter.java

```diff
@@ -703,6 +702,7 @@ public boolean hasNext() {
 
     @Override
     public void loadNext() throws IOException {
+      initializeNumRecords();
       while (!current.hasNext() && !iterators.isEmpty()) {
         current = iterators.remove();
       }
```

Comment:
Instead of calling `this.hasNext()`, we can extract the same lines into an inline function? e.g. `nextIterator`.

Reply:
Make sense. Thank you for the comment